pkg/transaction/payload.go

```diff
@@ -0,0 +1,180 @@
+package transaction
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+
+	"github.com/icon-project/stacks-go-sdk/internal/utils"
+	"github.com/icon-project/stacks-go-sdk/pkg/c32"
+	"github.com/icon-project/stacks-go-sdk/pkg/clarity"
+	"github.com/icon-project/stacks-go-sdk/pkg/stacks"
+)
+
+type Payload interface {
+	Serialize() ([]byte, error)
+	Deserialize([]byte) (int, error)
+}
+
+type TokenTransferPayload struct {
+	Recipient clarity.ClarityValue // Can be either StandardPrincipal or ContractPrincipal
+	Amount    uint64
+	Memo      string
+}
+
+type ContractCallPayload struct {
```

Comment:
Curios here, shouldn't there be a value option contracCallPayload so it can be called and send some gas tokens?

Reply:
I think gas tokens aren't necessary in the payload because Clarity can determine costs statically. See https://github.com/stacksgov/sips/blob/main/sips/sip-002/sip-002-smart-contract-language.md#measuring-transaction-costs-for-fee-collection

I use their fee estimation API in the broadcast transaction functions, which I think is for that purpose.